### DISCUSSION TITLE
Failed to execute `mapAsync` on `GPUBuffer`

### DISCUSSION THREAD

**Discussion by hittjwX** [brianpetro/obsidian-smart-connections#746] [Q&A] [answered]
I'm getting the following errors in Smart Connect. 
![image](https://github.com/user-attachments/assets/58d95358-7c98-4429-92a4-2a355d17cfdc)

Does this mean my GPU is failing to do something? 

I have a NIVIDIA GeForce RTX 2080.

![image](https://github.com/user-attachments/assets/79f6d4c5-240c-4ce6-96b2-3ea500c989d5)

Embedding is slow sometimes, but faster when Smart Connect is running.

Are GPU compatibilities in the model or the Smart Connect?

**Comment by brianpetro** [comment]
Interesting, I haven't seen that error before.

I'm getting ready to ship an update that will better utilize the GPU for embedding even without the Smart Connect app running (though, offloading to Smart Connect will likely still improve general performance in Obsidian).

The update uses a newer version of huggingface's transformers.js, which might sort out that issue. I'm still doing some testing but I might have the update ready for the early-release version (v2.2) tonight.

🌴

**Reply by hittjwX** [reply]
I'd be happy to test if you put test objectives in your Issues or release notes. Over the years, I've been a systems administrator and project manager for software development. More recently, I've done control testing.

**Comment by hittjwX** [comment]
Smart Connections v2.1.99 is using the GPU now. 

When chat and files stopped working, I switched to `TaylorAI/bge-micro-v2`, which blazes 9000 tokens a second before dropping to 4447, which is great. 

I'm getting a peak 47% utilization of my NVIDIA GeForce RTX 2080, which is better than what I get for Adobe Premiere.

Now, I'm troubleshooting why chat and files stopped working. I also don't see OpenRouter in there anymore, and it won't let me select a model for chat.

**Reply by brianpetro** [reply, marked as answer]
@jwhco if you re-open the settings after a few moments the chat model selection _should_ reappear. See https://github.com/brianpetro/obsidian-smart-connections/issues/750#issuecomment-2324466546 🌴

**Reply by hittjwX** [reply]
Everything is back on track. Chat and files work again. I'm going to restart everything and switch back to a higher fidelity embed.

**Comment by christianngarcia** [comment]
Thanks for following up. 😊
